### DISCUSSION TITLE
Login with more than 2 identity

### DISCUSSION THREAD

**Discussion by RayhanYulanda** [keycloak/keycloak#14111] [Q&A]
In keycloak we can use email or username to login but what I need is to login with identity ID, employee ID or email. Is it possible to make it without change the core?

**Comment by Aleson95** [comment]
You can create an Authentication SPI to override the default authentication flow to fit your need

**Reply by RayhanYulanda** [reply]
So we need to change the core right? There is no option to add it in UI?

**Reply by Aleson95** [reply]
Its not changing the core but extending the keycloak functionality.
Then in Keycloak Admin Console UI you can choose to change the authentication flow to use Authentication SPI that you have built and deployed.
Link reference:
[Extending the server] (https://www.keycloak.org/docs/latest/server_development/index.html#_extensions)
[Authentication SPI] (https://www.keycloak.org/docs/latest/server_development/index.html#_auth_spi)

**Reply by RayhanYulanda** [reply]
Thank you! It helps me a lot!
